### PR TITLE
GroupGallery Map

### PR DESCRIPTION
**Pull request by D0nPiano**
Added a map to GroupGallery. Might make sense for #877. Looks fairly different on mobile, since there were some trubbles with the footer and z-index.

ToDos:
- [x] Add Markers
- [x] Code Cleanup

map

### CODE REVIEW

**Comment by D0nPiano** [comment]
More changes than expected... :smile:  Should finally be done for now though!

Looks like this now:
### Desktop
map1
map2

### Mobile
mapmobile

### Possible To Dos (for other PRs)
- [ ] make group map use new "full width" slot in main Layout
- [ ] if user moved the map / zoomed, filter groups accordingly
- [ ] if user moved the map / zoomed, filter groups accordingly
- [ ] But the actual cards into a seperate component, rename GroupCards to GroupCardsLayout
- [ ] Enable searching for locations
- [ ] Find a way to make popups on map more pretty

**Comment by tiltec** [comment]
Wow, great changes! Looks good! A few nitpicks:

image

This could be aligned better, also I would prefer the arrow-up/arrow-down that we use in the sidebar instead of "x/+" icons.

image

Three group cards next to each other look a bit cramped on my laptop, I guess it's just too small. But something that could be improved.

File: src/components/GroupJoin/GroupGalleryCardsUI.vue [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            class="underline"
+          >
+            {{ $t('JOINGROUP.LOGOUT_MESSAGE.LOG_IN') }}
+          </router-link>
+        </i18n>
+      </q-alert>
+    </transition>
+    <transition name="slide-toggle">
+      <div
+        v-if="(expanded && filteredMyGroups.length>0) || (filteredOtherGroups.length == 0 && filteredMyGroups.length == 1)"
```
Comment (tiltec) [inline src/components/GroupJoin/GroupGalleryCardsUI.vue]: The condition is getting quite long, better make it a `computed` value.

File: src/components/GroupJoin/GroupGalleryCardsUI.vue [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      <q-card v-if="!previewOpened && !showMyGroups">
+        <q-search
+          @change="$emit('search', arguments[0])"
+          class="searchbar"
+          v-model="search"
+        />
+      </q-card>
+    </transition>
+    <transition name="slide-toggle">
+      <div v-if="(expanded && filteredOtherGroups.length > 0) || (filteredOtherGroups.length == 1 && filteredMyGroups.length == 0)">
```
Comment (tiltec) [inline src/components/GroupJoin/GroupGalleryCardsUI.vue]: < insert comment about complicated logic in templates here > ;)

Comment (D0nPiano) [reply src/components/GroupJoin/GroupGalleryCardsUI.vue]: Haha :laughing:

**Comment by tiltec** [comment]
This also needs consideration how to handle inactive groups (per #915 ). I added a quick fix to hide inactive groups from the list in #925, but maybe if someone explicitly searches for a group it should still show it? Inactive groups will anyway get shown if someone searches in the topbar.

No blocking issue, I'd merge this PR without filtering inactive groups.

For the testing, I'm not sure if I would like to merge without tests. Complicated logic tends to break easily on future edits and this can slow down development. I had quite a bad experience when I tried to edit the `PickupUsers` component (the one that calculates the number of slots depending on page width) and after a day of work, I gave up and did only the most minimal change. Might be my own stupidity, but maybe I would have been more successful if I had started with writing tests for it.

**Comment by D0nPiano** [comment]
Wrote a couple of tests now - especially for expanded / not expanded and for clicking on the preview button and the according change (not showing the other cards anymore, creating a GroupPreview component)

I saw the active field in /api/groups - but there are no inactive groups yet, right? Is there a way to make one inactive for testing purposes yet? Would also just wait a bit otherwise then...

Hm, yeah, that does sound annoying with the pickups - what did you want to change?

**Comment by tiltec** [comment]
> I saw the active field in /api/groups - but there are no inactive groups yet, right? Is there a way to make one inactive for testing purposes yet?

True. We would have to mark them in the database, there is no API planned for that. Also I want to rewrite the `active` field into a `status` field to accomodate the playground group too -> #914 
So we rather merge this and add the inactive & playground handling later.

> Hm, yeah, that does sound annoying with the pickups - what did you want to change?

 I fixed a bug where v-for would receive a negative number to iterate over. While doing that, I wanted to move some of the logic from the template into computed properties to make the component easier to debug.

**Comment by tiltec** [comment]
Merging this will break the end-to-end test that runs before deploying to production, just a reminder to myself to adapt the test soon.

**Pull request merged**